great_expectations/core/batch.py

```diff
@@ -223,39 +221,39 @@ def data_asset_name(self, data_asset_name) -> None:
         self._data_asset_name = data_asset_name
 
     @property
-    def data_connector_query(self) -> dict:
+    def data_connector_query(self) -> Union[dict, None]:
         return self._data_connector_query
 
     @data_connector_query.setter
     def data_connector_query(self, value: dict) -> None:
         self._data_connector_query = value
 
     @property
-    def limit(self) -> int:
+    def limit(self) -> Union[int, None]:
         return self._limit
 
     @limit.setter
     def limit(self, value: int) -> None:
         self._limit = value
 
     @property
-    def runtime_parameters(self) -> dict:
+    def runtime_parameters(self) -> Union[dict, None]:
         return self._runtime_parameters
 
     @runtime_parameters.setter
     def runtime_parameters(self, value: dict) -> None:
         self._runtime_parameters = value
 
     @property
-    def batch_identifiers(self) -> dict:
+    def batch_identifiers(self) -> Optional[dict]:
```

Comment:
Inconsistent use of `Optional[<TYPE>]` and `Union[<TYPE>, None]` here 😅 
Should update this.